src/Libs/GObject-2.0/Public/Object.Registration.cs

```diff
@@ -31,7 +33,25 @@ private Type GetGTypeOrRegister(System.Type type)
     private static class FallbackRegistrationStrategy
     {
         private static string QualifyName(System.Type type)
-            => type.FullName?.Replace(".", "") ?? type.Name;
+        {
+            if (type.IsGenericType)
```

Comment:
This looks a bit to complex for me. I think to fix this I think it should be possible to keep the idea of the original code (using simple `string.Replace`) but apply it to `Type.ToString()` instead of `Type.FullName`.

Something along the lines:
```CS
type.ToString()
    .Replace(".", string.Empty)
    .Replace("+", string.Empty)
    .Replace(",", "_")
    .Replace("`", string.Empty)
    .Replace("[", "_")
    .Replace("]", string.Empty);
```